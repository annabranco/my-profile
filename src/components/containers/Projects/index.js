import React, { useEffect } from 'react';
import { arrayOf, func, string } from 'prop-types';
import chunk from 'lodash/chunk';
import sortBy from 'lodash/sortBy';
import { useStateWithLabel } from '../../../utils/hooks';
import { isDesktop } from '../../../utils/device';
import { ProjectsList } from '../../views';
import {
  ACTUAL_PAGE,
  ADVANCE_ACTION,
  BACK_ACTION,
  PROJECTS_STATE,
  SHOW_THUMBNAILS_ACTION,
  SHOW_THUMBNAILS_ON_MOBILE_ACTION,
  THUMBS_STATE
} from '../../../constants';
import { developerTextPropType, projectsPropType } from '../../../types';

const Projects = ({
  adjustScrollAfterThumbnails,
  language,
  projects,
  texts
}) => {
  const [actualPage, changePage] = useStateWithLabel(1, ACTUAL_PAGE);
  const [projectsState, setProjectsState] = useStateWithLabel(
    {
      projects: null,
      totalPages: 0
    },
    PROJECTS_STATE
  );
  const [thumbsState, updateThumbsState] = useStateWithLabel(
    {
      adjustedView: 0,
      displayThumbnails: false
    },
    THUMBS_STATE
  );

  const organizeProjectsList = numberOfProjectsPerPage => {
    const pagedProjects = chunk(sortBy(projects), numberOfProjectsPerPage);
    setProjectsState({
      projects: pagedProjects,
      totalPages: pagedProjects.length
    });
  };

  const toggleProjectsThumbNails = isVisible => {
    updateThumbsState(prevState => ({
      adjustedView: isVisible
        ? prevState.adjustedView + 150
        : prevState.adjustedView - 150,
      displayThumbnails: isVisible
    }));
    adjustScrollAfterThumbnails(150);
  };

  const onClickChangePage = action => {
    let nextPage;

    if (action === ADVANCE_ACTION) {
      nextPage = actualPage + 1;
    } else if (action === BACK_ACTION) {
      nextPage = actualPage - 1;
    }

    if (nextPage) {
      changePage(nextPage);
    }
  };

  const getThumbnailsStyle = () => {
    if (thumbsState.displayThumbnails && isDesktop) {
      return SHOW_THUMBNAILS_ACTION;
    }
    if (thumbsState.displayThumbnails) {
      return SHOW_THUMBNAILS_ON_MOBILE_ACTION;
    }
    return '';
  };

  useEffect(() => {
    organizeProjectsList(isDesktop ? 2 : 1);
  }, []);

  return (
    <ProjectsList
      actualPage={actualPage}
      displayThumbnails={thumbsState.displayThumbnails}
      language={language}
      onClickChangePage={onClickChangePage}
      projects={projectsState.projects}
      texts={texts}
      thumbnailsStyle={getThumbnailsStyle()}
      toggleProjectsThumbNails={toggleProjectsThumbNails}
      totalPages={projectsState.totalPages}
    />
  );
};

Projects.propTypes = {
  adjustScrollAfterThumbnails: func.isRequired,
  language: string.isRequired,
  projects: arrayOf(projectsPropType).isRequired,
  texts: developerTextPropType.isRequired
};

export default Projects;
