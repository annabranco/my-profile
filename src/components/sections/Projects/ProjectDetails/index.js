import React from 'react';
import { useSelector } from 'react-redux';
import { number } from 'prop-types';
import { currentLanguageSelector } from '../../../../redux/selectors';
import { isDesktop } from '../../../../utils/device';
import { THUMBNAILS } from '../../../../assets/images';
import { projectsPropType } from '../../../../types';
import {
  Description,
  Link,
  ProjectItem,
  RepoIcon,
  RepoUrl,
  SeeIcon,
  Thumbnail,
  Title
} from './styles';

const ProjectDetails = ({ actualPage, project }) => {
  const languageSelected = useSelector(currentLanguageSelector);

  return (
    <ProjectItem actualPage={actualPage} key={project.title}>
      {isDesktop && (
        <>
          <Link href={project.url} rel="noopener noreferrer" target="_Blank">
            <Title>
              {project.title}
              <SeeIcon className="far fa-eye" />
            </Title>
          </Link>
          <Description>{project.description[languageSelected]}</Description>
          <Link href={project.repo} rel="noopener noreferrer" target="_Blank">
            <RepoUrl className="project__repo">
              <RepoIcon className="fab fa-github-alt" />
              {project.repo.replace('https://github.com', '')}
            </RepoUrl>
          </Link>
        </>
      )}
      <Link
        href={isDesktop ? project.url : project.repo}
        rel="noopener noreferrer"
        target="_Blank"
      >
        <Thumbnail src={THUMBNAILS[project.thumbnail]} alt={project.title} />
        {!isDesktop && <Title>{project.title}</Title>}
      </Link>
    </ProjectItem>
  );
};

ProjectDetails.propTypes = {
  actualPage: number.isRequired,
  project: projectsPropType.isRequired
};

export default ProjectDetails;
