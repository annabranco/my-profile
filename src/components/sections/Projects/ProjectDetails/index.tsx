import { ReactElement } from 'react';
import { useSelector } from 'react-redux';
import { number } from 'prop-types';
import { IProjects } from '../../../../types/interfaces';
import { currentLanguageSelector } from '../../../../redux/selectors';
import { isDesktop } from '../../../../utils/device';
import { THUMBNAILS } from '../../../../assets/images';
import { projectsPropType } from '../../../../types/propTypes';
import {
  Description,
  Link,
  ProjectItem,
  RepoIcon,
  RepoUrl,
  SeeIcon,
  Thumbnail,
  Title
} from './styles';

interface Props {
  actualPage: number;
  project: IProjects;
}

const ProjectDetails = ({ actualPage, project }: Props): ReactElement => {
  const languageSelected = useSelector(currentLanguageSelector);

  return (
    <ProjectItem actualPage={actualPage} key={project.title}>
      {isDesktop && (
        <>
          <Link
            href={project.url}
            data-e2e-id={`projects-title_${project.title}`}
            rel="noopener noreferrer"
            target="_Blank"
          >
            <Title>
              {project.title}
              <SeeIcon className="far fa-eye" />
            </Title>
          </Link>
          <Description>{project.description[languageSelected]}</Description>
          <Link
            href={project.repo}
            data-e2e-id={`projects-repo_${project.title}`}
            rel="noopener noreferrer"
            target="_Blank"
          >
            <RepoUrl className="project__repo">
              <RepoIcon className="fab fa-github-alt" />
              {project.repo.replace('https://github.com', '')}
            </RepoUrl>
          </Link>
        </>
      )}
      <Link
        data-e2e-id={`projects-thumb_${project.title}`}
        href={isDesktop ? project.url : project.repo}
        rel="noopener noreferrer"
        target="_Blank"
      >
        <Thumbnail
          src={THUMBNAILS[project.thumbnail as keyof typeof THUMBNAILS]}
          alt={project.title}
        />
        {!isDesktop && <Title>{project.title}</Title>}
      </Link>
    </ProjectItem>
  );
};

ProjectDetails.propTypes = {
  actualPage: number.isRequired,
  project: projectsPropType.isRequired
};

export default ProjectDetails;
