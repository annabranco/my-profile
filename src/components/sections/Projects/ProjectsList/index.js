import React from 'react';
import { useSelector } from 'react-redux';
import { arrayOf, bool, func, number, oneOf } from 'prop-types';
import { developerTextSelector } from '../../../../redux/selectors';
import ProjectDetails from '../ProjectDetails';
import {
  ADVANCE_ACTION,
  BACK_ACTION,
  SHOW_THUMBNAILS_ACTION,
  SHOW_THUMBNAILS_ON_MOBILE_ACTION
} from '../../../../constants';
import { projectsPropType } from '../../../../types';
import { Icon, Paginator, ProjectsGrid, ProjectsSection, Text } from './styles';
import { isDesktop } from '../../../../utils/device';

const ProjectsList = ({
  actualPage,
  onClickChangePage,
  projects,
  thumbnailsStyle,
  totalPages,
  visible
}) => {
  const texts = useSelector(developerTextSelector);

  return (
    <ProjectsSection>
      {(visible || !isDesktop) && (
        <>
          <ProjectsGrid thumbnailsStyle={thumbnailsStyle}>
            {projects &&
              projects[actualPage - 1].map(project => (
                <ProjectDetails
                  actualPage={actualPage}
                  key={project.title}
                  project={project}
                />
              ))}
          </ProjectsGrid>

          <Paginator>
            <Text notVisible={actualPage === 1}>{texts.goUp}</Text>
            <Icon
              aria-label={texts.goUp}
              className="far fa-arrow-alt-circle-up"
              notVisible={actualPage === 1}
              onClick={() => onClickChangePage(BACK_ACTION)}
              role="button"
              tabIndex={0}
            />
            <Icon
              aria-label={texts.showMore}
              className="far fa-arrow-alt-circle-down"
              next
              notVisible={actualPage === totalPages}
              onClick={() => onClickChangePage(ADVANCE_ACTION)}
              role="button"
              tabIndex={0}
            />
            <Text next notVisible={actualPage === totalPages}>
              {texts.showMore}
            </Text>
          </Paginator>
        </>
      )}
    </ProjectsSection>
  );
};

ProjectsList.propTypes = {
  actualPage: number.isRequired,
  onClickChangePage: func.isRequired,
  projects: arrayOf(arrayOf(projectsPropType)),
  thumbnailsStyle: oneOf([
    SHOW_THUMBNAILS_ON_MOBILE_ACTION,
    SHOW_THUMBNAILS_ACTION
  ]),
  totalPages: number.isRequired,
  visible: bool.isRequired
};

ProjectsList.defaultProps = {
  projects: [
    [
      {
        description: {
          en: '',
          es: '',
          pt: ''
        },
        order: 0,
        repo: '',
        thumbnail: '',
        url: ''
      }
    ]
  ],
  thumbnailsStyle: ''
};

export default ProjectsList;
