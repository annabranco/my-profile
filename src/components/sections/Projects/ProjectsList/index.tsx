import React, { ReactElement } from 'react';
import { useSelector } from 'react-redux';
import { arrayOf, bool, func, number } from 'prop-types';
import { IProjects } from '../../../../types/interfaces';
import { developerTextSelector } from '../../../../redux/selectors';
import { isDesktop } from '../../../../utils/device';
import ProjectDetails from '../ProjectDetails';
import { ADVANCE_ACTION, BACK_ACTION } from '../../../../constants';
import { projectsPropType } from '../../../../types/propTypes';
import { Icon, Paginator, ProjectsGrid, ProjectsSection, Text } from './styles';

interface Props {
  actualPage: number;
  onClickChangePage: (action: string) => void;
  projects: IProjects[][];
  totalPages: number;
  visible: boolean;
}

const ProjectsList = ({
  actualPage,
  onClickChangePage,
  projects,
  totalPages,
  visible
}: Props): ReactElement => {
  const texts = useSelector(developerTextSelector);

  return (
    <ProjectsSection>
      {(visible || !isDesktop) && (
        <>
          <ProjectsGrid>
            {projects &&
              projects[actualPage - 1].map(project => (
                <ProjectDetails
                  actualPage={actualPage}
                  key={project.title}
                  project={project}
                />
              ))}
          </ProjectsGrid>

          <Paginator>
            <Text notVisible={actualPage === 1}>{texts.goUp}</Text>
            <Icon
              aria-label={texts.goUp}
              className="far fa-arrow-alt-circle-up"
              notVisible={actualPage === 1}
              onClick={() => onClickChangePage(BACK_ACTION)}
              role="button"
              tabIndex={0}
            />
            <Icon
              aria-label={texts.showMore}
              className="far fa-arrow-alt-circle-down"
              next
              notVisible={actualPage === totalPages}
              onClick={() => onClickChangePage(ADVANCE_ACTION)}
              role="button"
              tabIndex={0}
            />
            <Text next notVisible={actualPage === totalPages}>
              {texts.showMore}
            </Text>
          </Paginator>
        </>
      )}
    </ProjectsSection>
  );
};

ProjectsList.propTypes = {
  actualPage: number.isRequired,
  onClickChangePage: func.isRequired,
  projects: arrayOf(arrayOf(projectsPropType)),
  totalPages: number.isRequired,
  visible: bool.isRequired
};

ProjectsList.defaultProps = {
  projects: [
    [
      {
        description: {
          en: '',
          es: '',
          pt: ''
        },
        order: 0,
        repo: '',
        thumbnail: '',
        url: ''
      }
    ]
  ]
};

export default ProjectsList;
