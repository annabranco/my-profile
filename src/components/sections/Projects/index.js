import React, { useEffect } from 'react';
import { useSelector } from 'react-redux';
import chunk from 'lodash/chunk';
import sortBy from 'lodash/sortBy';
import { projectsSelector } from '../../../redux/selectors';
import { useStateWithLabel } from '../../../utils/hooks';
import { isDesktop } from '../../../utils/device';
import ProjectsList from './ProjectsList';
import {
  ACTUAL_PAGE,
  ADVANCE_ACTION,
  BACK_ACTION,
  PROJECTS,
  SHOW_THUMBNAILS_ACTION,
  SHOW_THUMBNAILS_ON_MOBILE_ACTION
} from '../../../constants';

const Projects = () => {
  const projectsLoaded = useSelector(projectsSelector);

  const [actualPage, changePage] = useStateWithLabel(1, ACTUAL_PAGE);
  const [projectsState, setProjectsState] = useStateWithLabel(
    {
      projects: null,
      totalPages: 0
    },
    PROJECTS
  );

  const onClickChangePage = action => {
    let nextPage;

    if (action === ADVANCE_ACTION) {
      nextPage = actualPage + 1;
    } else if (action === BACK_ACTION) {
      nextPage = actualPage - 1;
    }

    if (nextPage) {
      changePage(nextPage);
    }
  };

  const getThumbnailsStyle = () =>
    isDesktop ? SHOW_THUMBNAILS_ACTION : SHOW_THUMBNAILS_ON_MOBILE_ACTION;

  useEffect(() => {
    const organizeProjectsList = numberOfProjectsPerPage => {
      const pagedProjects = chunk(
        sortBy(projectsLoaded),
        numberOfProjectsPerPage
      );
      setProjectsState({
        projects: pagedProjects,
        totalPages: pagedProjects.length
      });
    };

    organizeProjectsList(2);
  }, [projectsLoaded, setProjectsState]);

  return (
    <ProjectsList
      actualPage={actualPage}
      onClickChangePage={onClickChangePage}
      projects={projectsState.projects}
      thumbnailsStyle={getThumbnailsStyle()}
      totalPages={projectsState.totalPages}
    />
  );
};

export default Projects;
