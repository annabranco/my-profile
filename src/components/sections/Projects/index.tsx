import React, { ReactElement, useEffect } from 'react';
import { useSelector } from 'react-redux';
import { chunk, sortBy } from 'lodash';
import { instanceOf } from 'prop-types';
import { IProjectsIndexator, IProjects } from '../../../types/interfaces';
import { projectsSelector } from '../../../redux/selectors';
import { useStateWithLabel } from '../../../utils/hooks';
import ProjectsList from './ProjectsList';
import {
  ACTUAL_PAGE,
  ADVANCE_ACTION,
  BACK_ACTION,
  PROJECTS
} from '../../../constants';

interface Props {
  cuePointsActivated: Set<string>;
}

const Projects = ({ cuePointsActivated }: Props): ReactElement => {
  const projectsLoaded: IProjects[] = useSelector(projectsSelector);

  const [actualPage, changePage] = useStateWithLabel<number>(1, ACTUAL_PAGE);
  const [
    projectsState,
    setProjectsState
  ] = useStateWithLabel<IProjectsIndexator>(
    {
      projects: [],
      totalPages: 0
    },
    PROJECTS
  );

  const onClickChangePage = (action: string): void => {
    let nextPage;

    if (action === ADVANCE_ACTION) {
      nextPage = actualPage + 1;
    } else if (action === BACK_ACTION) {
      nextPage = actualPage - 1;
    }

    if (nextPage) {
      changePage(nextPage);
    }
  };

  useEffect(() => {
    const organizeProjectsList = (numberOfProjectsPerPage: number): void => {
      const pagedProjects = chunk(
        sortBy(projectsLoaded),
        numberOfProjectsPerPage
      );
      setProjectsState({
        projects: pagedProjects,
        totalPages: pagedProjects.length
      });
    };

    organizeProjectsList(2);
  }, [projectsLoaded, setProjectsState]);

  return (
    <ProjectsList
      actualPage={actualPage}
      onClickChangePage={onClickChangePage}
      projects={projectsState.projects}
      totalPages={projectsState.totalPages}
      visible={cuePointsActivated.has(PROJECTS)}
    />
  );
};

Projects.propTypes = {
  cuePointsActivated: instanceOf(Set).isRequired
};

export default Projects;
