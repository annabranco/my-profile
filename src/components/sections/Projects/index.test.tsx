import { shallow, ShallowWrapper } from 'enzyme';
import 'jest-styled-components';
import '../../../../testing/mocks/matchMedia';
import Projects from '.';
import { setupMockProvider } from '../../../../testing/mocks';
// import { getMockState } from '../../../testing/mocks/mockState';

// TODO [07-May -21]: Define real test suits (Anna Branco)

describe('< Projects >', () => {
  let MockProvider;
  let wrapper: ShallowWrapper;

  beforeAll(() => {
    MockProvider = setupMockProvider({ language: 'en' }).MockProvider;
    wrapper = shallow(
      <MockProvider>
        <Projects cuePointsActivated={new Set()} />
      </MockProvider>
    );
  });

  it('should mount', () => {
    expect(wrapper).toBeDefined();
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
