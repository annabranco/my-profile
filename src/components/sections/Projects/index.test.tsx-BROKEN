import { mount } from 'enzyme';
import 'jest-styled-components';
import '../../../../testing/mocks/matchMedia';
import Projects from '.';
import { setupMockProvider } from '../../../../testing/mocks';
// import { getMockState } from '../../../testing/mocks/mockState';

describe('< Projects >', () => {
  let MockProvider;
  // let MOCK_STATE;
  let component;

  beforeAll(() => {
    MockProvider = setupMockProvider({ language: 'en' }).MockProvider;
    // MOCK_STATE = getMockState({ language: 'en' });

    component = mount(
      <MockProvider>
        <Projects adjustScrollAfterThumbnails={() => null} />
      </MockProvider>
    );
  });

  it('should mount', () => {
    expect(component).toBeDefined();
  });

  it('matches the snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
