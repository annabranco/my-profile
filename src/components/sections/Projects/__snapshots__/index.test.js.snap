// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< Projects > matches the snapshot 1`] = `
<MockProvider>
  <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Projects
      adjustScrollAfterThumbnails={[Function]}
    >
      <ProjectsList
        actualPage={1}
        displayThumbnails={false}
        onClickChangePage={[Function]}
        projects={
          Array [
            Array [
              Object {
                "description": Object {
                  "en": "This is the page you are currently visiting. It was created as a personal project developed with React.js as a digital presentation of my professional profile.",
                  "es": "Esta es la página donde estás navegando. Ha sido creada como un proyecto personal desarrollado con React.js para la presentación digital de mi perfil profesional.",
                  "pt": "Esta é a página onde você está. Foi criada como um projeto pessoal desenvolvido com React.js para apresentação digital do meu perfil profissional.",
                },
                "order": 1,
                "repo": "https://github.com/annabranco/my-profile",
                "thumbnail": "https://raw.githubusercontent.com/annabranco/my-profile/master/src/images/thumbnails/thumbnail-my-profile.jpg",
                "title": "My profile",
                "url": "https://annabranco.github.io/my-profile/",
              },
              Object {
                "description": Object {
                  "en": "A fun and beautiful memory card game themed on Disney’s Toy Story characters, for all ages. It is based on my final evaluation test from Adalab.",
                  "es": "Un divertido y bonito juego de parejas, con tema de Toy Story (Disney), para todas las edades. Está basado en mi ejercicio final de evaluación de Adalab.",
                  "pt": "Um divertido e bonito jogo da memória, com temática de Toy Story (Disney), para todas as idades. Foi inspirado em meu exercício final de avaliação de Adalab.",
                },
                "order": 2,
                "repo": "https://github.com/annabranco/toy-story-games",
                "thumbnail": "https://raw.githubusercontent.com/annabranco/my-profile/master/src/images/thumbnails/thumbnail-toy-story.jpg",
                "title": "Toy Story Games",
                "url": "https://annabranco.github.io/toy-story-games/",
              },
            ],
            Array [
              Object {
                "description": Object {
                  "en": "The protptype of an own Role Playing Game built in vanilla JavaScript. There is currently just one simple task to accomplish. The assets were not created by me.",
                  "es": "El prototipo de un juego de rol mío desarrollado en JavaScript vanilla. Actualmente hay solamente un objetivo por alcanzar. Los assets no fueron creados por mí.",
                  "pt": "O protótipo de um jogo de RPG meu programado en JavaScript vanilla. No momento só há um simples objetivo por alcançar. Os assets não foram criados por mim.",
                },
                "order": 3,
                "repo": "https://github.com/annabranco/an-adventure",
                "thumbnail": "https://raw.githubusercontent.com/annabranco/my-profile/master/src/images/thumbnails/thumbnail-an-adventure.jpg",
                "title": "An Adventure",
                "url": "https://annabranco.github.io/an-adventure/",
              },
              Object {
                "description": Object {
                  "en": "A prototype of a web page of football team. The news and contact pages are embed content from Wordpress to reproduce back-end functionalities.",
                  "es": "Un prototipo de una página web de un equipo de fútbol. La parte de noticias y contacto son contenidos externos reproduciendo funcionalidades back-end.",
                  "pt": "Um protótipo de uma página web de um time de futebol. A parte de notícias e contato são conteúdos externos reproduzindo funcionalidades back-end.",
                },
                "order": 4,
                "repo": "https://github.com/annabranco/nipiterfc",
                "thumbnail": "https://raw.githubusercontent.com/annabranco/my-profile/master/src/images/thumbnails/thumbnail-ni-piter.jpg",
                "title": "Ni Piter FC",
                "url": "https://annabranco.github.io/nipiterfc/",
              },
            ],
            Array [
              Object {
                "description": Object {
                  "en": "A beautiful TV series finder, where users can search on a huge database, also see all the series summaries, ratings and add them to favorites.",
                  "es": "Un buscador de series de televisión, donde los usuarios pueden buscar en una grande base de datos y ver sinopsis, valoraciones y añadir a favoritos.",
                  "pt": "Um buscador de séries de TV, onde os usuários podem buscar en um enorme banco de dados, bem como ver sinopses, avaliações e adicionar a favoritos.",
                },
                "order": 5,
                "repo": "https://github.com/annabranco/tv-series-finder",
                "thumbnail": "https://raw.githubusercontent.com/annabranco/my-profile/master/src/images/thumbnails/thumbnail-tv-series-finder.jpg",
                "title": "TV Series Finder",
                "url": "https://annabranco.github.io/tv-series-finder/",
              },
              Object {
                "description": Object {
                  "en": "The final technical project from Adalab developed by a group of five students (called Ohana). It generates a professional profile card to be shared.",
                  "es": "Proyecto técnico final de Adalab desarrollado por un equipo de 5 estudiantes (llamadas Ohana). Crea una tarjeta virtual profesional para compartirla.",
                  "pt": "Projeto técnico final de Adalab, desenvolvido por um time de 5 estudantes (chamadas Ohana). Cria um cartão virtual profissional para ser compartilhado.",
                },
                "order": 6,
                "repo": "https://github.com/Adalab/dorcas-s3-ohana-group",
                "thumbnail": "https://raw.githubusercontent.com/annabranco/my-profile/master/src/images/thumbnails/thumbnail-awesome-profile-cards.jpg",
                "title": "Awesome Profile Cards",
                "url": "http://beta.adalab.es/dorcas-s3-ohana-group/",
              },
            ],
            Array [
              Object {
                "description": Object {
                  "en": "A beautiful visual database of Harry Potter characters, with filters and favorites function. My future objective is to develop the database as my own API.",
                  "es": "Un proyecto visual de los personajes de Harry Potter, con filtros y gestión de favoritos. Mi futuro objetivo es desarrollar todos los datos en una API mía. ",
                  "pt": "Un bonito projeto visual dos personagens de Harry Potter, com filtros e gestão de favoritos. Meu objetivo futuro é desenvolver os dados em minha própria API.",
                },
                "order": 7,
                "repo": "https://github.com/annabranco/harry-potter-database",
                "thumbnail": "https://raw.githubusercontent.com/annabranco/my-profile/master/src/images/thumbnails/thumbnail-harry-potter-database.jpg",
                "title": "Harry Potter Database",
                "url": "https://annabranco.github.io/harry-potter-database",
              },
            ],
          ]
        }
        thumbnailsStyle={null}
        toggleProjectsThumbNails={[Function]}
        totalPages={4}
      >
        <PROJECTS Section>
          <section
            className="dsRSxI"
          >
            <Checkbox Area>
              <div
                className="hBpPhe"
              >
                <Checkbox
                  id="developer__projects-checkbox"
                  onClick={[Function]}
                  type="checkbox"
                >
                  <input
                    className="gMgTPJ"
                    id="developer__projects-checkbox"
                    onClick={[Function]}
                    type="checkbox"
                  />
                </Checkbox>
                <label
                  htmlFor="developer__projects-checkbox"
                >
                  Show thumbnails
                </label>
              </div>
            </Checkbox Area>
            <Projects Grid
              thumbnailsStyle={null}
            >
              <ul
                className="hUHje"
              >
                <ProjectDetails
                  displayThumbnails={false}
                  key="My profile"
                  project={
                    Object {
                      "description": Object {
                        "en": "This is the page you are currently visiting. It was created as a personal project developed with React.js as a digital presentation of my professional profile.",
                        "es": "Esta es la página donde estás navegando. Ha sido creada como un proyecto personal desarrollado con React.js para la presentación digital de mi perfil profesional.",
                        "pt": "Esta é a página onde você está. Foi criada como um projeto pessoal desenvolvido com React.js para apresentação digital do meu perfil profissional.",
                      },
                      "order": 1,
                      "repo": "https://github.com/annabranco/my-profile",
                      "thumbnail": "https://raw.githubusercontent.com/annabranco/my-profile/master/src/images/thumbnails/thumbnail-my-profile.jpg",
                      "title": "My profile",
                      "url": "https://annabranco.github.io/my-profile/",
                    }
                  }
                >
                  <Project Item
                    key="My profile"
                  >
                    <div
                      className="jhrGlQ"
                    >
                      <Link
                        href="https://annabranco.github.io/my-profile/"
                        rel="noopener noreferrer"
                        target="_Blank"
                      >
                        <a
                          className="kOEPgv"
                          href="https://annabranco.github.io/my-profile/"
                          rel="noopener noreferrer"
                          target="_Blank"
                        >
                          <Title>
                            <h3
                              className="PEspk"
                            >
                              My profile
                              <SeeIcon
                                className="far fa-eye"
                              >
                                <i
                                  className="kdhGNa far fa-eye"
                                />
                              </SeeIcon>
                            </h3>
                          </Title>
                        </a>
                      </Link>
                      <Description>
                        <p
                          className="jVJjFT"
                        >
                          This is the page you are currently visiting. It was created as a personal project developed with React.js as a digital presentation of my professional profile.
                        </p>
                      </Description>
                      <Link
                        href="https://github.com/annabranco/my-profile"
                        rel="noopener noreferrer"
                        target="_Blank"
                      >
                        <a
                          className="kOEPgv"
                          href="https://github.com/annabranco/my-profile"
                          rel="noopener noreferrer"
                          target="_Blank"
                        >
                          <p
                            className="project__repo"
                          >
                            <Repo Icon
                              className="fab fa-github-alt"
                            >
                              <i
                                className="iUKfTa fab fa-github-alt"
                              />
                            </Repo Icon>
                            /annabranco/my-profile
                          </p>
                        </a>
                      </Link>
                    </div>
                  </Project Item>
                </ProjectDetails>
                <ProjectDetails
                  displayThumbnails={false}
                  key="Toy Story Games"
                  project={
                    Object {
                      "description": Object {
                        "en": "A fun and beautiful memory card game themed on Disney’s Toy Story characters, for all ages. It is based on my final evaluation test from Adalab.",
                        "es": "Un divertido y bonito juego de parejas, con tema de Toy Story (Disney), para todas las edades. Está basado en mi ejercicio final de evaluación de Adalab.",
                        "pt": "Um divertido e bonito jogo da memória, com temática de Toy Story (Disney), para todas as idades. Foi inspirado em meu exercício final de avaliação de Adalab.",
                      },
                      "order": 2,
                      "repo": "https://github.com/annabranco/toy-story-games",
                      "thumbnail": "https://raw.githubusercontent.com/annabranco/my-profile/master/src/images/thumbnails/thumbnail-toy-story.jpg",
                      "title": "Toy Story Games",
                      "url": "https://annabranco.github.io/toy-story-games/",
                    }
                  }
                >
                  <Project Item
                    key="Toy Story Games"
                  >
                    <div
                      className="jhrGlQ"
                    >
                      <Link
                        href="https://annabranco.github.io/toy-story-games/"
                        rel="noopener noreferrer"
                        target="_Blank"
                      >
                        <a
                          className="kOEPgv"
                          href="https://annabranco.github.io/toy-story-games/"
                          rel="noopener noreferrer"
                          target="_Blank"
                        >
                          <Title>
                            <h3
                              className="PEspk"
                            >
                              Toy Story Games
                              <SeeIcon
                                className="far fa-eye"
                              >
                                <i
                                  className="kdhGNa far fa-eye"
                                />
                              </SeeIcon>
                            </h3>
                          </Title>
                        </a>
                      </Link>
                      <Description>
                        <p
                          className="jVJjFT"
                        >
                          A fun and beautiful memory card game themed on Disney’s Toy Story characters, for all ages. It is based on my final evaluation test from Adalab.
                        </p>
                      </Description>
                      <Link
                        href="https://github.com/annabranco/toy-story-games"
                        rel="noopener noreferrer"
                        target="_Blank"
                      >
                        <a
                          className="kOEPgv"
                          href="https://github.com/annabranco/toy-story-games"
                          rel="noopener noreferrer"
                          target="_Blank"
                        >
                          <p
                            className="project__repo"
                          >
                            <Repo Icon
                              className="fab fa-github-alt"
                            >
                              <i
                                className="iUKfTa fab fa-github-alt"
                              />
                            </Repo Icon>
                            /annabranco/toy-story-games
                          </p>
                        </a>
                      </Link>
                    </div>
                  </Project Item>
                </ProjectDetails>
              </ul>
            </Projects Grid>
            <Paginator>
              <div
                className="ftYEKY"
              >
                <Text
                  hidden={true}
                >
                  <span
                    className="cmtkEf"
                    hidden={true}
                  >
                    go up
                  </span>
                </Text>
                <Icon
                  aria-label="go up"
                  className="far fa-arrow-alt-circle-up"
                  hidden={true}
                  onClick={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  <i
                    aria-label="go up"
                    className="iaRxNl far fa-arrow-alt-circle-up"
                    hidden={true}
                    onClick={[Function]}
                    role="button"
                    tabIndex={0}
                  />
                </Icon>
                <Icon
                  aria-label="show more"
                  className="far fa-arrow-alt-circle-down"
                  hidden={false}
                  onClick={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  <i
                    aria-label="show more"
                    className="iaRxNl far fa-arrow-alt-circle-down"
                    hidden={false}
                    onClick={[Function]}
                    role="button"
                    tabIndex={0}
                  />
                </Icon>
                <Text
                  hidden={false}
                >
                  <span
                    className="cmtkEf"
                    hidden={false}
                  >
                    show more
                  </span>
                </Text>
              </div>
            </Paginator>
          </section>
        </PROJECTS Section>
      </ProjectsList>
    </Projects>
  </Provider>
</MockProvider>
`;
